### DISCUSSION TITLE
Improve "Charging Cost Stats" dashboard - Free charged panels

### DISCUSSION THREAD

**Discussion by denzel1994** [CarlosCuezva/dashboards-Grafana-Teslamate#30] [General]
Hey,

on the regular version of the "Charging Cost Stats" dashboard the two "Free charged in total" panels show a zero.
This happens, because all my free charges have no cost defined. 

For those panels to output the correct value also for charges with no defined price I had to change:
`AND cost = 0`
to
`AND (cost = 0 OR cost isnull)`

**Comment by CarlosCuezva** [comment]
Hello

A null price cannot be considered a free recharge

**Reply by denzel1994** [reply]
Hello, okay thank you. :)

Then I'll have to enter a 0 cost for all relevant charges / geofences.

Another improvement would be to differentiate free used and added kWh:
<img width="207" alt="Bildschirmfoto 2023-11-28 um 12 28 19" src="https://github.com/CarlosCuezva/dashboards-Grafana-Teslamate/assets/36551243/cc9efa89-dfde-4d0f-b101-f53514371490">

**Reply by CarlosCuezva** [reply]
Ok, I'll see it.

I'll let you know with anything

**Reply by CarlosCuezva** [reply]
Something similar to this?
![image](https://github.com/CarlosCuezva/dashboards-Grafana-Teslamate/assets/5815421/b4501cfa-3f83-43e8-a7ce-5f08fce1f34d)
